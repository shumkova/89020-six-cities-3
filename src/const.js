export const AppRoute = {
  LOGIN: `/login`,
  ROOT: `/`,
  FAVORITES: `/favorites`,
  OFFER: `/offer`,
};

export const AppState = {
  PENDING: `Pending`,
  READY: `Ready`,
};

export const ListKind = {
  OFFER: `offer`,
  NEAR: `near`,
};

export const SortTypes = {
  POPULAR: `Popular`,
  PRICE_TO_HIGH: `Price: low to high`,
  PRICE_TO_LOW: `Price: high to low`,
  TOP_RATED: `Top rated first`,
};

export const ListTypes = {
  CITY: {
    list: `cities__places-list`,
    card: `cities__place-card`,
  },
  NEARBY: {
    list: `near-places__list`,
    card: `near-places__card`,
  },
  FAVORITE: {
    list: ``,
    card: `favorites__card`,
  },
};
