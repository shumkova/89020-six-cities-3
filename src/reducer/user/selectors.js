import NameSpace from "../name-space.js";

const NAME_SPACE = NameSpace.USER;


const getAuthorizationStatus = (state) => {
  return state[NAME_SPACE].authorizationStatus;
};

const getAuthInfo = (state) => {
  return state[NAME_SPACE].authInfo;
};

export {getAuthorizationStatus, getAuthInfo};

