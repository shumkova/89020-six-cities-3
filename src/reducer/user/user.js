import {extend} from "../../utils";
import EditAuthInfo from "../../adapters/edit-auth-info";
import {AppRoute} from "../../const";
import history from "../../history";

const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

const initialState = {
  authorizationStatus: AuthorizationStatus.NO_AUTH,
  authInfo: {},
};

const ActionTypes = {
  REQUIRED_AUTHORIZATION: `REQUIRED_AUTHORIZATION`,
  SET_INFO: `SET_INFO`,
};

const ActionCreator = {
  requireAuthorization: (status) => {
    return {
      type: ActionTypes.REQUIRED_AUTHORIZATION,
      payload: status,
    };
  },

  setUser: (data) => {
    return {
      type: ActionTypes.SET_INFO,
      payload: data,
    };
  }
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionTypes.REQUIRED_AUTHORIZATION:
      return extend(state, {
        authorizationStatus: action.payload,
      });

    case ActionTypes.SET_INFO:
      return extend(state, {
        authInfo: action.payload,
      });
  }

  return state;
};

const Operation = {
  checkAuth: () => (dispatch, getState, api) => {
    return api.get(`/login`)
      .then((response) => {
        dispatch(ActionCreator.requireAuthorization(AuthorizationStatus.AUTH));
        dispatch(ActionCreator.setUser(EditAuthInfo.parseUser(response.data)));
      })
      .catch((err) => {
        throw err;
      });
  },

  login: (authData) => (dispatch, getState, api) => {
    return api.post(`/login`, {
      email: authData.login,
      password: authData.password,
    })
      .then((response) => {
        const user = EditAuthInfo.parseUser(response.data);
        dispatch(ActionCreator.requireAuthorization(AuthorizationStatus.AUTH));
        dispatch(ActionCreator.setUser(user));
        history.push(AppRoute.ROOT);
      });
  }
};

export {
  ActionCreator,
  ActionTypes,
  AuthorizationStatus,
  Operation,
  reducer,
};
