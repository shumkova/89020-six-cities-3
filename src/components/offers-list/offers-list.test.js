import React from "react";
import renderer from "react-test-renderer";
import OffersList from "./offers-list.connect";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import NameSpace from "../../reducer/name-space";
import {Router} from "react-router-dom";
import history from "../../history";
import {AuthorizationStatus} from "../../reducer/user/user";
import {ListTypes, SortTypes} from "../../const";

const mockStore = configureStore([]);

const offers = [{
  bedrooms: 3,
  city: {
    location: {
      latitude: 52.370216,
      longitude: 4.895168,
      zoom: 10
    },
    name: `Amsterdam`
  },
  description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
  host: {
    avatar: `img/1.png`,
    id: 3,
    isPro: true,
    name: `Angelina`
  },
  id: 1,
  images: [`img/1.png`, `img/2.png`],
  isFavorite: false,
  isPremium: false,
  location: {
    latitude: 52.35514938496378,
    longitude: 4.673877537499948,
    zoom: 8
  },
  adults: 4,
  preview: `img/1.png`,
  price: 120,
  rating: 4.8,
  title: `Beautiful & luxurious studio at great location`,
  type: `apartment`
}];

const noop = () => {};

it(`Render OffersList`, () => {
  const store = mockStore({
    [NameSpace.DATA]: {
      hotels: offers,
      applicationIsReady: true,
    },
    [NameSpace.APP]: {
      city: `Amsterdam`,
      activeOffer: null,
      currentOffer: null,
      reviews: [],
      nearbyOffers: [],
      sortType: SortTypes.POPULAR,
      reviewLoadingStatus: ``,
    },
    [NameSpace.USER]: {
      authorizationStatus: AuthorizationStatus.NO_AUTH,
      authInfo: {},
    }
  });

  const tree = renderer.create(
      <Router history={history}>
        <Provider store={store}>
          <OffersList
            offers={offers}
            onHeaderClick={noop}
            onBookmarkClick={noop}
            onItemHover={noop}
            listType={ListTypes.CITY}
            setActiveOffer={noop}
          />
        </Provider>
      </Router>

  ).toJSON;

  expect(tree).toMatchSnapshot();
});
