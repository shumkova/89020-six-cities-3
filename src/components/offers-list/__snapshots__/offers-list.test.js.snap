// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render OffersList 1`] = `[Function]`;
